docs/Contributing/Fleet-UI-Testing.md

```diff
@@ -0,0 +1,375 @@
+# Fleet UI testing
+
+This document contains the testing strategy and plan for the frontend codebase. The testing
+strategy is a high-level overview of the **who**, **what**, **when** and **why** when it comes to testing.
+The testing plan primarily outlines the **how** of testing and covers the different practices and
+toolings used in testing.
+
+For instructions on using our testings tools, check out our [testing docs](../Contributing/Testing-and-local-development.md).
+
+**Table of contents**
+
+- [Testing strategy](#testing-strategy)
+  - [Testing philosophy](#testing-philosophy)
+  - [Who tests](#who-tests)
+  - [What we test](#what-we-test)
+    - [Shared utilities](#shared-utilities)
+    - [UI building blocks](#ui-building-blocks-resusable-components-and-hooks)
+    - [App widgets](#app-widgets)
+    - [User journeys](#user-journeys)
+- [Testing plan](#testing-plan)
+  - [Types of tests](#types-of-tests)
+    - [Testing cheat sheet](#testing-cheat-sheet)
+    - [Manual testing](#manual-testing)
+    - [Static analysis](#static-analysis)
+    - [Unit testing](#unit-testing)
+      - [Shared utilities testing](#shared-utilities-testing)
+      - [UI building block testing](#ui-building-block-testing)
+      - [Reusable hook testing](#reusable-hooks-testing)
+    - [Integration testing](#integration-testing)
+      - [App widgets testing](#app-widget-testing)
+    - [E2E testing](#e2e-testing)
+  - [Tooling](#tooling)
+    - [ESLint and TypeScript](#eslint-and-typescript)
+    - [Jest](#jest)
+    - [Testing Library](#testing-library)
```

Comment:
```suggestion
    - [Testing library](#testing-library)
```
If this is referring to a testing library in general, then it is not a proper noun and should follow sentence case in this context. The Cypress Testing Library is a library with a particular name, which is why it is capitalized in this manner.